### PR TITLE
SWATCH-2958: Set billing producer aws when role is moa

### PR DESCRIPTION
**Pull request by Sgitario**
Jira issue: SWATCH-2958

## Description
If the product label on the telemetry feed is (moa-hostedcontrolplane|moa), then force the billing_marketplace to aws.

## Testing
Added junit test to reproduce the issue.

### CODE REVIEW

File: swatch-model-events/schemas/event.yaml [hunk 1]
```diff
@@ -186,6 +186,7 @@ properties:
       - addon-open-data-hub
       - BASILISK
       - moa-hostedcontrolplane
+      - moa
```
Comment (Sgitario) [inline swatch-model-events/schemas/event.yaml:189]: As far as I see, we're using an enum Role to parse from string, and then to parse it back to String. 
I think we should remove this Role enum and directly use a string here. wdyt ? @kahowell

Comment (kflahert) [reply swatch-model-events/schemas/event.yaml:189]: Thinking we should go with a similar idea as what we did with ProductId. We created a class that only allows values that come directly from our config: https://github.com/RedHatInsights/rhsm-subscriptions/blob/main/swatch-product-configuration/src/main/java/com/redhat/swatch/configuration/registry/ProductId.java

Not required for this fix so I'm approving and can look again if we make updates.

**Review by kflahert — APPROVED** [review]

File: swatch-model-events/schemas/event.yaml (hunk 1 above)
Comment (Sgitario) [reply swatch-model-events/schemas/event.yaml:189]: I've created https://issues.redhat.com/browse/SWATCH-3002 for futher discussion.

**Comment by Aurobinda55** [comment]
Looks good to me, 
Added moa-hostedcontrolplane|moa events with billing_provider="". 
swatch replaced billing_provider=aws

**Review by Aurobinda55 — APPROVED** [review]

**Pull request merged**